catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserDefinedBrickListFragment.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui.fragment
+
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.ListFragment
+import org.catrobat.catroid.R
+import org.catrobat.catroid.content.bricks.Brick
+import org.catrobat.catroid.content.bricks.UserDefinedBrick
+import org.catrobat.catroid.databinding.FragmentUserDefinedBrickListBinding
+import org.catrobat.catroid.ui.adapter.PrototypeBrickAdapter
+import org.catrobat.catroid.ui.fragment.AddBrickFragment.OnAddBrickListener
+import org.catrobat.catroid.utils.ToastUtil
+
+class UserDefinedBrickListFragment : ListFragment(), View.OnClickListener {
+
+    private var addBrickListener: OnAddBrickListener? = null
+    private var adapter: PrototypeBrickAdapter? = null
+
+    private var _binding: FragmentUserDefinedBrickListBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        val activity = activity as AppCompatActivity?
+        val actionBar = activity?.supportActionBar
+        actionBar?.setTitle(R.string.categories)
```

Comment:
better safer code would be
```suggestion
        requireActivity()
            .takeIf { it is AppCompatActivity }
            .let { it as AppCompatActivity }
            .supportActionBar
            ?.setTitle(R.string.categories)
```
since this is used again in OnCreateView line 68, please export it as a function and used it.